pinecone/core/grpc/index_grpc.py

```diff
@@ -155,42 +160,160 @@ def wrapped():
     def _request_metadata(self) -> Dict[str, str]:
         return {REQUEST_ID: _generate_request_id()}
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.close()
+
+
+class vector(object):
+    def __init__(self, id: str, values: list, metadata: dict):
+        self.id = id
+        self.values = values
+        self.metadata = metadata
+
+    def __repr__(self):
+        return str({'id': self.id, 'values': self.values, 'metadata': self.metadata})
+
+
+class FetchResponse(object):
+    def __init__(self, vectors: dict, namespace: str):
+        self.vectors = vectors
+        self.namespace = namespace
+
+    def __repr__(self):
+        return str({'vectors': self.vectors, 'namespace': self.namespace})
+
+
+def parse_fetch_response(response: dict):
+    vd = {}
+    vectors = response.get('vectors')
+    if not vectors:
+        return None
+    for id, vec in vectors.items():
+        v_obj = vector(id=vec['id'], values=vec['values'], metadata=vec.get('metadata', None))
+        vd[id] = v_obj
+    namespace = response.get('namespace', None)
+    f = FetchResponse(vectors=vd, namespace=namespace)
+    return f
+
+
+class ScoredVector(object):
+    def __init__(self, id: str, score: float, values: list, metadata: dict):
+        self.id = id
+        self.score = score
+        self.values = values
+        self.metadata = metadata
 
-class Index(GRPCIndex):
+    def __repr__(self):
+        return str({'id': self.id, 'score': self.score, 'values': self.values, 'metadtata': self.metadata})
+
+
+class QueryResult(object):
+    def __init__(self, matches: list, namespace: str):
+        self.matches = matches
+        self.namespace = namespace
+
+    def __repr__(self):
+        return str({'matches': self.matches, 'namespace': self.namespace})
+
+
+class QueryResponse(object):
```

Comment:
The openapi client model objects allows accessing fields like:
- resp.get('results', default_value) 
- resp['results']
- resp.results

I guess what you have is just the resp.results style? Maybe that's fine as long as we don't show usage of the other styles anywhere in our docs (I think we don't, but could you double check in core/api-docs/code_samples?

Also keep in mind adding pydoc for these classes - that's something that we'll want soon, but doesn't have to be this PR.